src/main/scala/firrtl/transforms/InlineBooleanExpressions.scala

```diff
@@ -144,26 +145,33 @@ class InlineBooleanExpressions extends Transform with DependencyAPIMigration {
   }
 
   def execute(state: CircuitState): CircuitState = {
-    val dontTouchMap: Map[OfModule, Set[Ref]] = {
-      val refTargets = state.annotations.flatMap {
-        case anno: HasDontTouches => anno.dontTouches
-        case o => Nil
-      }
-      val dontTouches: Seq[(OfModule, Ref)] = refTargets.map {
-        case r => Target.referringModule(r).module.OfModule -> r.ref.Ref
+    val run = !state.annotations.contains(PrettyNoExprInlining)
```

Comment:
I think the option and its overall behavior is great.

My main concern is with the approach of "run the transform and do nothing" vs. "don't run the transform." I think flags like this are a good opportunity to make better use of dependencies (i.e. making sure that optimizations are only ever in the transitive `optionalPrerequisite` closure of non-optimization transforms and removing optimizations from the stage's requested targets).

I don't think this PR is necessarily the right time or place, but it would be great to move toward solving these pass-ordering and pass-enabling problems with dependencies.